scripts/scheduled_posts.coffee

```diff
@@ -15,21 +15,18 @@ switch process.env.NODE_ENV
 # Connect to database
 db = mongojs(process.env.MONGOHQ_URL, ['articles'])
 
-db.articles.update(
-  { scheduled_publish_at: { $lt: new Date() } }
-  { 
-    $set: {
-      published: true
-      published_at: Date()
-      scheduled_publish_at: null 
-    } 
-  }
-  { multi: true }
-  (err, msg) ->
-    return exit err if err
-    debug 'Scheduled publication finished', msg
+db.articles.find(
+  scheduled_publish_at: { $lt: new Date() }
+).forEach (err, doc) ->
+  if !doc
+    debug 'Scheduled publication finished'
     process.exit()
-)
+  if err
+    exit err
+  doc.published = true
+  doc.published_at = moment(doc.scheduled_publish_at).toDate()
+  doc.scheduled_publish_at = null
+  db.articles.save doc
```

Comment:
Also food for thought: Given the non-blocking nature of Node we could also just wrap this script in a function call and use setInterval (essentially what [node-cron](https://github.com/ncb000gt/node-cron) does):

````
setInterval publishScheduled, 10000
````

To run this every 10 seconds (or w/e tighter interval) and that might allow using straight `update` call. I'm not recommending that—just wanted to point out how you can think about asynchronous operations outside of the typical request-response code.